### PR TITLE
SqlScript: Store queries in configuration properties fix

### LINKED ISSUE #719: SqlScript: Store queries in configuration properties
**Details of the scenario you tried and the problem that is occurring:**
I believe xSQLServerScript should behave in a manner consistent with the [DSC Script resource](https://docs.microsoft.com/en-us/powershell/dsc/scriptResource) where the contents of the Get/Set/Test scripts are stored in the configuration, not in an external file. I do see the value in being able to execute the Get/Set/Test from external SQL files, so I believe that functionality should be retained.

I propose the following properties be added to the resource:
- GetScript
- SetScript
- TestScript

This will be consistent with the DSC Script resource. The logic of the resource should be updated to allow any combination of the Get*, Set*, and Test* properties (eg. GetScript, TestScript, SetFilePath), but throw an error if both Get*, Set*, or Test* properties are set. This would be easiest with parameter sets, however that functionality is not available yet.

**The DSC configuration that is using the resource (as detailed as possible):**
```
Configuration ExampleSqlScript
{
    param
    (
        [Parameter(Mandatory = $true)]
        [PSCredential]
        $WindowsCredential,

        [Parameter(Mandatory = $false)]
        [System.String]
        $SaName = 'RenamedSa'
    )
    
    xSQLServerScript 'ChangeSaName'
    {
        ServerInstance = 'localhost\SQL2016'
        GetScript = 'SELECT Name FROM sys.server_principals WHERE principal_id = 1'
        SetScript = "
            DECLARE @saName sysname
            DECLARE @alterUser varchar(max)

            SELECT @saName = Name
            FROM sys.server_principals
            WHERE principal_id = 1

            SET @alterUser = 'ALTER LOGIN [' + @saName + '] WITH Name = [$SaName]'

            EXEC sp_executesql @alterUser
        "
        TestScript = "
            SELECT
                CASE
                    WHEN Name = '$SaName' THEN 1 
                    ELSE 0 
                END 
            FROM sys.server_principals
            WHERE principal_id = 1
        "
        PsDscRunAsCredential = $WindowsCredential
    }
}
```

**Version of the Operating System, SQL Server and PowerShell the DSC Target Node is running:**
N/A

**What module (SqlServer or SQLPS) and which version of the module the DSC Target Node is running:**
N/A

**Version of the DSC module you're using, or 'dev' if you're using current dev branch:**
v8.0.0.0/dev

### PR DESCRIPTION
**Pull request by chasewilson**
#### Pull Request (PR) description
Adds the ability to use Strings as queries for the SqlScript resource instead of just file paths to sql scripts

#### This Pull Request (PR) fixes the following issues
Fixes #719

#### Task list

- [x] Added an entry under the Unreleased section in the CHANGELOG.md? Entry
      should say what was changed, and how that affects users (if applicable).
- [ ] Resource documentation added/updated in README.md?
- [ ] Resource parameter descriptions added/updated in README.md, schema.mof
      and comment-based help?
- [x] Comment-based help added/updated?
- [x] Localization strings added/updated in all localization files as appropriate?
- [x] Examples appropriately added/updated?
- [x] Unit tests added/updated?
      See [DSC Resource Testing Guidelines](https://github.com/PowerShell/DscResources/blob/master/TestsGuidelines.md).
- [x] Integration tests added/updated (where possible)?
      See [DSC Resource Testing Guidelines](https://github.com/PowerShell/DscResources/blob/master/TestsGuidelines.md).
- [x] New/changed code adheres to
      [DSC Resource Style Guidelines](https://github.com/PowerShell/DscResources/blob/master/StyleGuidelines.md)
      and [Best Practices](https://github.com/PowerShell/DscResources/blob/master/BestPractices.md)?

---
This change is [Reviewable](https://reviewable.io/reviews/powershell/sqlserverdsc/1107)

### CODE REVIEW

**Comment by johlju** [comment]
@chasewilson thank you for starting to fix this issue! :smile: 

---

Review status: 0 of 6 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---

*[DSCResources/MSFT_SqlScript/MSFT_SqlScript.schema.mof, line 4 at r1](https://beta.reviewable.io/reviews/powershell/sqlserverdsc/1107#-L9kgcEy-W6rnZykSQIL:-L9kgcEy-W6rnZykSQIM:b-9vzmw8) ([raw file](https://github.com/powershell/sqlserverdsc/blob/4deb29c4b361d1e2b95343c666ade1061160505f/DSCResources/MSFT_SqlScript/MSFT_SqlScript.schema.mof#L4)):*
> ```
> class MSFT_SqlScript : OMI_BaseResource
> {
>     [Key, Description("The name of an instance of the Database Engine. For a default instance, only specify the computer name. For a named instance, use the format ComputerName\\InstanceName")] String ServerInstance;
> ```

We have a logical error here that we need to solve. When only this parameter is key, only one SqlScript resource can be used for each instance in the same resource.

See the examples that are failing in the tests here as an example:
https://ci.appveyor.com/project/PowerShell/sqlserverdsc/build/9.0.270.0?fullLog=true#L1446

Do you have any suggestion on how we can solve this?

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/powershell/sqlserverdsc/1107#-:-L9khIlc2AzLcvC14AtR:buqhin1)*

**Comment by johlju** [comment]
@chasewilson Let me know if you need any help or want to discuss this change. Really would like to see us finding a solution so this can be merged 😄

**Comment by chasewilson** [comment]
@johlju I would love to work on this issue with you! Sorry about the response time, I had to replace my machine and getting it set up again was a pain;p

**Comment by chasewilson** [comment]
Ok , my though tis there are a couple of things we could do here. We could create another key value in the Schema MOF and not actually use it in the configuration. SecurityPolicyDsc actually does this in its MSFT_AccountPolicy resource.

Or, I could create a second SqlScript Resource and name it SqlScriptQuery or something along those lines and keep the key values as the GetFilePath in the SqlScriptResource.

What are your thoughts on it?

**Comment by johlju** [comment]
@chasewilson No worries, we all have a lot going on. 😄 
I actually had the same thoughts. Add a new "dummy" property as a Key, or add a new resource. I actually in favor of the idea of creating a new resource, if you want to go down that road I'm all for it! You probably can copy unit and integration tests as well and just make minor changes for them to use the new resource.

**Comment by johlju** [comment]
Ignore the tests, they failed on downloading the SQL Server media - the build worker sometimes fail on this. I will review and if there aren't any problems I will kick off the tests again.

**Comment by johlju** [comment]
Are this PR ready for review again? 😃

**Comment by chasewilson** [comment]
I think it should be ready:)

**Comment by johlju** [comment]
Awesome work on this one! :smile:  I will continue the review once the first review comment is resolved. :slightly_smiling_face: 

---

Review status: 0 of 8 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[a discussion](https://beta.reviewable.io/reviews/powershell/sqlserverdsc/1107#-LAby-av5hUFOObP9IAs:-LAby-av5hUFOObP9IAt:b-ej7ruo) (no related file):*
The change log does not match the one in the dev branch. I think you need to rebase you working branch using `git rebase` (not using `git pull` or `git merge`). If you don't know how to rebase, no worries, I will help you, and there is a guide here; https://github.com/PowerShell/DscResources/blob/master/GettingStartedWithGitHub.md#resolve-merge-conflicts

---

*[CHANGELOG.md, line 4 at r2](https://beta.reviewable.io/reviews/powershell/sqlserverdsc/1107#-LAbwfgbCawg_VilYzwJ:-LAbwfgc98FO0j0lkbVi:b3hfuzh) ([raw file](https://github.com/powershell/sqlserverdsc/blob/985162fd8423abf01db3d2b1cc060054caee396f/CHANGELOG.md#L4)):*
> ```Markdown
> 
> ## Unreleased
> - Changes to SqlScriptQuery
> ```

We don't have to specify that we added unit test and integration tests, they are a package with the resource, so maybe just write this as follows? 

- Changes to SqlServerDsc
  - [Chase Wilson (@chasewilson)](https://github.com/chasewilson): Added new resource SqlScriptQuery

*The existing entries in the change log for unit and integration tests are because they have be altered.*

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/powershell/sqlserverdsc/1107#-:-LAbyPJUCSA6cGoBofPc:bjbj09x)*

**Comment by chasewilson** [comment]
@johlju ok, Sorry about that! :/ I messed up my rebase last night but I should have it fixed now:) I really appreciate your help and working through this with me!

**Comment by johlju** [comment]
No worries, it's gets easier each time. :) I am happy to help.

Sorry to say that there are still problem in the change file, and GitHub sees CHANGELOG.md as a conflicting file :) But no worries, if you do the steps I wrote above that will help you resolve this. Let me know if you get stuck. 🙂

**Comment by chasewilson** [comment]
I had rebased and merged, did I mess it up again @johlju

**Comment by johlju** [comment]
@chasewilson No worries, you are not messing up anything. 🙂 I just realized I didn't write that you should do `git push <remote-name-to-fork> dev --force`  at the end. Sorry, my bad 😞 
If you run `git remote -v` you see which remote name points to your fork.

If it doesn't work this time, please post the output from  `git remote -v` and I will see if I'm giving you the wrong instructions.

**Comment by chasewilson** [comment]
No worries!:) I keep seeing that it reaches it so I assumed it would be complete. I push to the remote branch and get an access denied error. (remote: Permission to PowerShell/SqlServerDsc.git denied to chasewilson.)

here is my git remote output 
origin  https://github.com/chasewilson/SqlServerDsc.git (fetch)
origin  https://github.com/chasewilson/SqlServerDsc.git (push)
upstream        https://github.com/PowerShell/SqlServerDsc.git (fetch)
upstream        https://github.com/PowerShell/SqlServerDsc.git (push)

**Comment by johlju** [comment]
You get the access denied because you are not allowed to push directly to 'PowerShell\SqlServerDsc'. What you need to do is to push to your fork `git push origin dev --force`. 
When you update your branch on your fork, then GitHub will automatically update the PR in 'PowerShell\SqlServerDsc'.

**Comment by chasewilson** [comment]
Oh, I've already done that and my changes were pushed over to here, my last commit was to add an example after a rebase and merge

**Comment by johlju** [comment]
There are still conflicting files (see below) so the rebase doesn't take. We will sort this out. 😃 
Giving you more complete steps, let me know if any step is unclear and I will explain it in more detail. Sorry if they are to detailed.

Note: Before you run `git push origin dev --force` you can always revert by cloning your fork again. 

```
git checkout dev
git fetch upstream dev
git rebase upstream/dev
# Fix any merge conflicts in the files using for example VS Code. 
# After that you only save the file. Do not commit the changes with a commit message.
git add *
git rebase --continue
# Fix any merge conflicts in the files using for example VS Code. 
# After that you only save the file. Do not commit the changes with a commit message.
git add *
git rebase --continue
# ... etc until there are no more merge conflicts.
git push origin dev --force
```

It should look something like this (cloned your fork and run this test) - this is from the step `git rebase upstream/dev` and only shows one `git rebase --continue`, you might need to do many until the rebase is finsihed.

```
PS > git rebase upstream/dev
First, rewinding head to replay your work on top of it...
Applying: updated tests
Applying: updates SqlScript Resource to accept queries
Applying: adds integration tests
Applying: updates to add error handling for multiple param inputs
Applying: updates error handling to standards and adds localized data folder
Applying: updates Changelog
error: Failed to merge in the changes.
Using index info to reconstruct a base tree...
M       CHANGELOG.md
Falling back to patching base and 3-way merge...
Auto-merging CHANGELOG.md
CONFLICT (content): Merge conflict in CHANGELOG.md
Patch failed at 0006 updates Changelog
The copy of the patch that failed is found in: .git/rebase-apply/patch

Resolve all conflicts manually, mark them as resolved with
"git add/rm <conflicted_files>", then run "git rebase --continue".
You can instead skip this commit: run "git rebase --skip".
To abort and get back to the state before "git rebase", run "git rebase --abort".

PS > git add *
PS > git rebase --continue
Applying: updates Changelog
Applying: updates change log to fix merg conflict
error: Failed to merge in the changes.
Using index info to reconstruct a base tree...
M       CHANGELOG.md
Falling back to patching base and 3-way merge...
Auto-merging CHANGELOG.md
CONFLICT (content): Merge conflict in CHANGELOG.md
Patch failed at 0007 updates change log to fix merg conflict
The copy of the patch that failed is found in: .git/rebase-apply/patch

Resolve all conflicts manually, mark them as resolved with
"git add/rm <conflicted_files>", then run "git rebase --continue".
You can instead skip this commit: run "git rebase --skip".
To abort and get back to the state before "git rebase", run "git rebase --abort".
```

And in VS Code it should look like this for each merge conflict.

image

**Comment by chasewilson** [comment]
Thanks for being patient! I promise I will get there eventually :p Ok, so I hope that works better, Everything seemed to go along with the instructions. Didn't have any merge merges and finished al merge conflicts then did a push to my dev.

**Comment by johlju** [comment]
No worries, always happy to help out! 😃  It looks like the rebase took this time! 😃 A good sign it's working is when all the commits move to the end of the PR.  I will review again!

**Comment by johlju** [comment]
@chasewilson could you please go into Reviewable and write 'Done' (or click the little Done-button) on each review comment. Then I know you are done, and I can also acknowledge the review comments which resolves the review comments. 

You find Reviewable in the big purple button in the PR description. In Reviewable you just write 'Done' on each comment (or discuss the comment) and when you written a comment on each review comment then you click the big green Publish button on the top of the page. That will send all the comments back to GitHub as one comment.

**Comment by chasewilson** [comment]
Done. It wouldn't work in edge and I couldn't figure out why it wasn't publishing so I moved to Chrome for the time being

---

Review status: 0 of 14 files reviewed at latest revision, 24 unresolved discussions, some commit checks failed.

---

*[a discussion](https://reviewable.io/reviews/powershell/sqlserverdsc/1107#-LBW9fib9ycqtuTztgQ_:-LBaxn3G50P3hldSajMs:b-8b11j9) (no related file):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Please add this new resource to the README.md (in alphabetical order).
</blockquote></details>

Added

---

*[CHANGELOG.md, line 4 at r2](https://reviewable.io/reviews/powershell/sqlserverdsc/1107#-LAbwfgbCawg_VilYzwJ:-LBaxsIk4OGxioj4Ntci:b-896fix) ([raw file](https://github.com/powershell/sqlserverdsc/blob/985162fd8423abf01db3d2b1cc060054caee396f/CHANGELOG.md#L4)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

We don't have to specify that we added unit test and integration tests, they are a package with the resource, so maybe just write this as follows? 

- Changes to SqlServerDsc
  - [Chase Wilson (@chasewilson)](https://github.com/chasewilson): Added new resource SqlScriptQuery

*The existing entries in the change log for unit and integration tests are because they have be altered.*

</blockquote></details>

Done.

---

*[CHANGELOG.md, line 15 at r6](https://reviewable.io/reviews/powershell/sqlserverdsc/1107#-LBW4zLn-Al7TZvWJA-q:-LBaxv763sloMpqIVij_:b-896fix) ([raw file](https://github.com/powershell/sqlserverdsc/blob/91e516dbb266cb342f342623fa8d43e05a6fe82a/CHANGELOG.md#L15)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

You have merge conflicts saved in this file, could you resolve these? Easiest is to see how the original change log looks like, and then add you entries correctly :slightly_smiling_face: 

https://github.com/PowerShell/SqlServerDsc/blob/dev/CHANGELOG.md#unreleased
</blockquote></details>

Done.

---

*[CHANGELOG.md, line 71 at r6](https://reviewable.io/reviews/powershell/sqlserverdsc/1107#-LBW5RsX5soIDvtiBiGg:-LBaxwoOA8JYaQ7p8UIs:b-896fix) ([raw file](https://github.com/powershell/sqlserverdsc/blob/91e516dbb266cb342f342623fa8d43e05a6fe82a/CHANGELOG.md#L71)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Here are something strange as well, I think your entries are here that should be moved to the Unreleased section? These entries under section 11.1.0.0 should remain unchanged. Please compare to the original change log and move any entries that you made.
</blockquote></details>

Done.

---

*[DSCResources/MSFT_SqlScript/MSFT_SqlScript.psm1, line 86 at r6](https://reviewable.io/reviews/powershell/sqlserverdsc/1107#-LBW6VAT8TzKjoY0RSzZ:-LBXqU3F9ilrwNa8InWL:bb7f2mf) ([raw file](https://github.com/powershell/sqlserverdsc/blob/91e516dbb266cb342f342623fa8d43e05a6fe82a/DSCResources/MSFT_SqlScript/MSFT_SqlScript.psm1#L86)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> ```
> "Stop"
> ```

Can we use single quotes here?
</blockquote></details>

done

---

*[DSCResources/MSFT_SqlScript/MSFT_SqlScript.psm1, line 184 at r6](https://reviewable.io/reviews/powershell/sqlserverdsc/1107#-LBW6cSf7moSxNBY_2Vu:-LBXrn_Y89bKKRoHDKLs:bb6un1j) ([raw file](https://github.com/powershell/sqlserverdsc/blob/91e516dbb266cb342f342623fa8d43e05a6fe82a/DSCResources/MSFT_SqlScript/MSFT_SqlScript.psm1#L184)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> ```
> "Stop"
> ```

Can we use single quotes here?
</blockquote></details>

Done

---

*[DSCResources/MSFT_SqlScript/MSFT_SqlScript.psm1, line 191 at r6](https://reviewable.io/reviews/powershell/sqlserverdsc/1107#-LBWAaS_63zp24uLn_-9:-LBXrvdp4Sjtn65IEU6e:bfao4yg) ([raw file](https://github.com/powershell/sqlserverdsc/blob/91e516dbb266cb342f342623fa8d43e05a6fe82a/DSCResources/MSFT_SqlScript/MSFT_SqlScript.psm1#L191)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

If we just write out the exception as a verbose message the resource will never fail. Is that the intended behavior? 
</blockquote></details>

Not sure why that was there.. I got it from somewhere but I don't think I meant to have it. Now it's just the default error if it fails and same for the next one

---

*[DSCResources/MSFT_SqlScript/MSFT_SqlScript.psm1, line 274 at r6](https://reviewable.io/reviews/powershell/sqlserverdsc/1107#-LBW6pRz9ajXsaOV05av:-LBa_y907Ci5I3HyDsfZ:b-896fix) ([raw file](https://github.com/powershell/sqlserverdsc/blob/91e516dbb266cb342f342623fa8d43e05a6fe82a/DSCResources/MSFT_SqlScript/MSFT_SqlScript.psm1#L274)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> ```
> "Stop"
> ```

Can we use single quotes here?
</blockquote></details>

Done.

---

*[DSCResources/MSFT_SqlScript/MSFT_SqlScript.psm1, line 326 at r6](https://reviewable.io/reviews/powershell/sqlserverdsc/1107#-LBW77bL1L9ky5-JM2jB:-LBaa-RN6lOqeNU_lY8r:b-896fix) ([raw file](https://github.com/powershell/sqlserverdsc/blob/91e516dbb266cb342f342623fa8d43e05a6fe82a/DSCResources/MSFT_SqlScript/MSFT_SqlScript.psm1#L326)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> ```
> ParameterSetName = 'File'
> ```

Can we set this back to `Mandatory = $true`?
</blockquote></details>

Done.

---

*[DSCResources/MSFT_SqlScript/MSFT_SqlScript.psm1, line 353 at r6](https://reviewable.io/reviews/powershell/sqlserverdsc/1107#-LBW7FN58zkAJEc9C1Sx:-LBaa0qDElX60wWGmjWm:b-896fix) ([raw file](https://github.com/powershell/sqlserverdsc/blob/91e516dbb266cb342f342623fa8d43e05a6fe82a/DSCResources/MSFT_SqlScript/MSFT_SqlScript.psm1#L353)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

This should not be need anymore? If we remove this, then we can remove the `-InputFile` from the `Invoke-SqlCmd` and splat that parameter as well?
</blockquote></details>

Done.

---

*[DSCResources/MSFT_SqlScript/MSFT_SqlScript.psm1, line 356 at r6](https://reviewable.io/reviews/powershell/sqlserverdsc/1107#-LBW7UZw8vnZlnPlo31T:-LBaa2jf7z6B0PLZ7Pld:b-896fix) ([raw file](https://github.com/powershell/sqlserverdsc/blob/91e516dbb266cb342f342623fa8d43e05a6fe82a/DSCResources/MSFT_SqlScript/MSFT_SqlScript.psm1#L356)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> ```
> -InputFile $InputFile
> ```

See comment above.
</blockquote></details>

Done.

---

*[DSCResources/MSFT_SqlScriptQuery/MSFT_SqlScriptQuery.psm1, line 42 at r6](https://reviewable.io/reviews/powershell/sqlserverdsc/1107#-LBWBNjN4havzDK_fmVX:-LBaa4KNC0A5TanE2hly:b-896fix) ([raw file](https://github.com/powershell/sqlserverdsc/blob/91e516dbb266cb342f342623fa8d43e05a6fe82a/DSCResources/MSFT_SqlScriptQuery/MSFT_SqlScriptQuery.psm1#L42)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> ```
> 'GetFilePath'
> ```

Should be GetQuery?
</blockquote></details>

Done.

---

*[DSCResources/MSFT_SqlScriptQuery/MSFT_SqlScriptQuery.psm1, line 86 at r6](https://reviewable.io/reviews/powershell/sqlserverdsc/1107#-LBWAJ17-2s1fZfdbjEx:-LBaa5Vq0syDDTTaTIsi:b-896fix) ([raw file](https://github.com/powershell/sqlserverdsc/blob/91e516dbb266cb342f342623fa8d43e05a6fe82a/DSCResources/MSFT_SqlScriptQuery/MSFT_SqlScriptQuery.psm1#L86)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> ```
> "Stop"
> ```

Can we use single quotes here?
</blockquote></details>

Done.

---

*[DSCResources/MSFT_SqlScriptQuery/MSFT_SqlScriptQuery.psm1, line 184 at r6](https://reviewable.io/reviews/powershell/sqlserverdsc/1107#-LBWABmw9Rf9D6ky6e18:-LBaaBEN-yqCAHt8t_qN:b-896fix) ([raw file](https://github.com/powershell/sqlserverdsc/blob/91e516dbb266cb342f342623fa8d43e05a6fe82a/DSCResources/MSFT_SqlScriptQuery/MSFT_SqlScriptQuery.psm1#L184)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> ```
> "Stop"
> ```

Can we use single quotes here?
</blockquote></details>

Done.

---

*[DSCResources/MSFT_SqlScriptQuery/MSFT_SqlScriptQuery.psm1, line 191 at r6](https://reviewable.io/reviews/powershell/sqlserverdsc/1107#-LBWAjto-cmYOYLaUhIf:-LBaaCcm6lEmzPy4Sa_d:b-896fix) ([raw file](https://github.com/powershell/sqlserverdsc/blob/91e516dbb266cb342f342623fa8d43e05a6fe82a/DSCResources/MSFT_SqlScriptQuery/MSFT_SqlScriptQuery.psm1#L191)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

If we just write out the exception as a verbose message the resource will never fail. Is that the intended behavior? 
</blockquote></details>

Done.

---

*[DSCResources/MSFT_SqlScriptQuery/MSFT_SqlScriptQuery.psm1, line 290 at r6](https://reviewable.io/reviews/powershell/sqlserverdsc/1107#-LBWAly67fi_QFe5XXqR:-LBaaEUkEZ9m0zHZO467:b-896fix) ([raw file](https://github.com/powershell/sqlserverdsc/blob/91e516dbb266cb342f342623fa8d43e05a6fe82a/DSCResources/MSFT_SqlScriptQuery/MSFT_SqlScriptQuery.psm1#L290)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

If we just write out the exception as a verbose message the resource will never fail. Is that the intended behavior? 
</blockquote></details>

Done.

---

*[DSCResources/MSFT_SqlScriptQuery/MSFT_SqlScriptQuery.psm1, line 318 at r6](https://reviewable.io/reviews/powershell/sqlserverdsc/1107#-LBW8u7v2A7ksHEIxoQZ:-LBZKjvH7dirojgEpzx5:b-ds9a22) ([raw file](https://github.com/powershell/sqlserverdsc/blob/91e516dbb266cb342f342623fa8d43e05a6fe82a/DSCResources/MSFT_SqlScriptQuery/MSFT_SqlScriptQuery.psm1#L318)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Could we move this helper function to the CommonResourceHelper.psm1 (and it's tests to CommonResourceHelper.Tests.ps1)? Then we can make sure that both SqlScript and SqlScriptQuery uses the the same helper function.
</blockquote></details>

Moved/combined the functions to the CommonResourceHelper as well as Removed/Moved/COmbined the Unit tests for it/them

---

*[DSCResources/MSFT_SqlScriptQuery/MSFT_SqlScriptQuery.schema.mof, line 10 at r6](https://reviewable.io/reviews/powershell/sqlserverdsc/1107#-LBWBER3C5SiCyWxk7Fz:-LBaaJTS-HhYXsWQnDd8:b-896fix) ([raw file](https://github.com/powershell/sqlserverdsc/blob/91e516dbb266cb342f342623fa8d43e05a6fe82a/DSCResources/MSFT_SqlScriptQuery/MSFT_SqlScriptQuery.schema.mof#L10)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> ```
> time out.  In some SQL
> ```

Got two spaces the after the '.'
</blockquote></details>

Done.

---

*[DSCResources/MSFT_SqlScriptQuery/MSFT_SqlScriptQuery.schema.mof, line 11 at r6](https://reviewable.io/reviews/powershell/sqlserverdsc/1107#-LBWBS8JCdQZa9tfUXec:-LBaaKaF9hJ5OYJ5jjPM:b-896fix) ([raw file](https://github.com/powershell/sqlserverdsc/blob/91e516dbb266cb342f342623fa8d43e05a6fe82a/DSCResources/MSFT_SqlScriptQuery/MSFT_SqlScriptQuery.schema.mof#L11)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>


> ```
> 'GetFilePath'
> ```

Should be GetQuery?
</blockquote></details>

Done.

---

*[Examples/Resources/SqlScriptQuery/1-RunScriptUsingSQLAuthentication.ps1, line 24 at r6](https://reviewable.io/reviews/powershell/sqlserverdsc/1107#-LBWB_jP63s91eF8reIv:-LBaaPPd2_9tCy8TRf92:b-896fix) ([raw file](https://github.com/powershell/sqlserverdsc/blob/91e516dbb266cb342f342623fa8d43e05a6fe82a/Examples/Resources/SqlScriptQuery/1-RunScriptUsingSQLAuthentication.ps1#L24)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Can we align the equal signs?
</blockquote></details>

Done.

---

*[Examples/Resources/SqlScriptQuery/2-RunScriptUsingWindowsAuthentication.ps1, line 23 at r6](https://reviewable.io/reviews/powershell/sqlserverdsc/1107#-LBWBhdmAuAzHkmT7WCF:-LBaaZT-5-c5QgpNybmu:b-896fix) ([raw file](https://github.com/powershell/sqlserverdsc/blob/91e516dbb266cb342f342623fa8d43e05a6fe82a/Examples/Resources/SqlScriptQuery/2-RunScriptUsingWindowsAuthentication.ps1#L23)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Can we align the equal signs? Throughout.
</blockquote></details>

Done.

---

*[Tests/Integration/MSFT_SqlScriptQuery.Integration.Tests.ps1, line 210 at r6](https://reviewable.io/reviews/powershell/sqlserverdsc/1107#-LBWD6O54-aNyW4vgIfc:-LBaacl4-dr_DPbazTBY:b-896fix) ([raw file](https://github.com/powershell/sqlserverdsc/blob/91e516dbb266cb342f342623fa8d43e05a6fe82a/Tests/Integration/MSFT_SqlScriptQuery.Integration.Tests.ps1#L210)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Please remove extra blank row.
</blockquote></details>

Done.

---

*[Tests/Unit/MSFT_SqlScript.Tests.ps1, line 73 at r6](https://reviewable.io/reviews/powershell/sqlserverdsc/1107#-LBW82iY5U4OSsRB7mLR:-LBaag286Ks3izjOXikC:b-896fix) ([raw file](https://github.com/powershell/sqlserverdsc/blob/6c087ae92ef6ec1f6c1886a8578cc407be16764f/Tests/Unit/MSFT_SqlScript.Tests.ps1#L73)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> ```
> Should -Throw
> ```

This has replaced all the Pester v4 syntax from `Should -Throw` to `Should Throw`. Can you please revert this?
</blockquote></details>

Done.

---

*[Tests/Unit/MSFT_SqlScript.Tests.ps1, line 85 at r6](https://reviewable.io/reviews/powershell/sqlserverdsc/1107#-LBW8HaL7bohi8o0TfmQ:-LBabAWe8EGUQdpiIzaC:b-896fix) ([raw file](https://github.com/powershell/sqlserverdsc/blob/6c087ae92ef6ec1f6c1886a8578cc407be16764f/Tests/Unit/MSFT_SqlScript.Tests.ps1#L85)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> ```
> Should -Be
> ```

This has replaced all the Pester v4 syntax from `Should -Be` to `Should Be`. Can you please revert this?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/powershell/sqlserverdsc/1107#-:-LBayE0iD-bd7b6lLVlL:b-bwrk0w)*

**Comment by chasewilson** [comment]
Ok fresh PR coming up and hopefully much easier! after making the changes, I'm running into an issue with the unit tests though. After moving the Invoke-SqlScript to the helper module when I run the unit tests on either SqlScript or SqlScriptQuery I get the following issue, 
Invoke-Sqlcmd: StubNotImplemented'. from C:\source2\SqlServerDsc\Tests\Unit\Stubs\SQLPSStub.psm1:606
I know where it's coming from but not entirely sure of the reasons for the stubs or how how they interact with the tests. Any help I can get is greatly appreciated @johlju  :)

**Comment by johlju** [comment]
The stubs are meant so that the unit tests can run on a dev machine that does not have SQL Server (and the SQLPS or SQLServer module). So the stubs is a empty version of the real cmdlet, and when called it throws `StubNotImplemented` to show that the cmdlet was wrongly called. The reason it's called is because there is a missing  `Mock -CommandName 'Invoke-Sqlcmd' ...` in the tests. The `Mock` command is probably run higher up where the tests initially came from. 
So basically, the stub throws so that you can see that a mock of the cmdlet was missed. 🙂

**Comment by chasewilson** [comment]
Ahh I forgot about the scoping of the mock! Mock is in the right place but moving it out of the sqlscript modules moved it out of the scope if the mock! Thank you!
